tests/model/test_files_bugzilla_map.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+
+from treeherder.model.models import BugzillaComponent, FilesBugzillaMap, Repository
+from treeherder.etl.files_bugzilla_map import FilesBugzillaMapProcess
+
+EXPECTED_PROJECTS = [
+    'mozilla-central',
+    'mozilla-beta',
+    'mozilla-release',
+    'mozilla-esr78',
+]
+
+
+@pytest.mark.django_db(transaction=True)
+def test_get_project_to_import(setup_repository_data):
+    """
+    Test for which projects the mapping of files to Bugzilla components shall be
+    imported and if the order is correct.
+    """
+    actual_projects = list(
+        Repository.objects.filter(codebase='gecko')
+        .filter(active_status='active')
+        .filter(life_cycle_order__isnull=False)
+        .values_list('name', flat=True)
+        .order_by('life_cycle_order')
```

Comment:
I'm not following the purpose of `life_cyle_order` other than to apply to certain gecko repo's and rank them? What's the ranking useful for?

Reply:
`life_cycle_order` determines the priority of the repository for the file to Bugzilla product+component mapping. A repository with a low value gets processed first and data for the same file path in repositories processed later will be ignored. This allows to require Bugzilla component reorganizations to only require changes on mozilla-central. The other sheriffed repositories need to be processed to have the data for files which got moved or deleted in mozilla-central.

Reply:
ok, thanks for the explanation.